src/adapter/src/coord/message_handler.rs

```diff
@@ -62,8 +62,13 @@ impl Coordinator {
         async move {
             match msg {
                 Message::Command(otel_ctx, cmd) => {
-                    otel_ctx.attach_as_parent();
-                    self.message_command(cmd).await
+                    // TODO: We need a Span that is not none for the otel_ctx to attach the parent
+                    // relationship to. If we swap the otel_ctx in `Command::Message` for a Span, we
+                    // can downgrade this to a debug_span.
+                    let span = tracing::info_span!("message_command").or_current();
+                    span.in_scope(|| otel_ctx.attach_as_parent());
+                    tracing::Span::current().add_link(span.context().span().span_context().clone());
```

Comment:
I think we don't need this? It got added at some point the old bit of code I'm resurrecting here, but I don't know why and I'm not entirely sure I understand it

Reply:
Verified locally that we don't seem to need it, removing